source/Condition.h

```diff
@@ -0,0 +1,194 @@
+/* Condition.h
+Copyright (c) 2022 by an anonymous author.
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef RVALUE_H_
+#define RVALUE_H_
+
+
+#include <cmath>
+#include <limits>
+#include <string>
+#include <type_traits>
+
+
+
+// This class stores either:
+//   1. A condition's value and key
+//   2. A literal value (key is empty)
+//
+// The value (V) is first in the template since (nearly?) all Condition
+// classes will use a std::string key (K). Having a template key type
+// allows storage of scope, wstring, etc. in the future without
+// rewriting this class.
+template < class V, class K = std::string >
+class Condition {
+public:
+	typedef V ValueType;
+	typedef K KeyType;
+
+	constexpr Condition() : value(), key() {}
+	explicit constexpr Condition(const V &value) : value(value), key() {}
+	constexpr Condition(const V &value, const K &key) : value(value), key(key) {}
+	~Condition() {}
+
+	template <class V2, class K2>
```

Comment:
dont those have to be the same V, K for this function to work?
mb its just a weird template thing

Reply:
No.

Reply:
idea is you can convert any of these to any other easily I guess
do we intend to convert them at all even?

Reply:
Yes. A `Condition<T>` has to be a drop-in replacement for T, otherwise any code that may need conditions will have to be rewritten. Since C++ allows implicit conversion of arithmetic types, a `Condition<arithmetic type>` must also allow implicit conversions.

Reply:
Thus far, the code needs Conditions that are unsigned, int, int64_t, and double.

Reply:
you can resolve this